File: data/EXP01/golang_no_system_var/2502/2502.output_2.go
type Allocator struct {
    memory []int
}

func Constructor(n int) Allocator {
    return Allocator{
        memory: make([]int, n),
    }
}

func (a *Allocator) Allocate(size int, mID int) int {
    for i := 0; i <= len(a.memory)-size; i++ {
        if a.isMemoryFree(i, size) {
            a.assignMemory(i, size, mID)
            return i
        }
    }
    return -1
}

func (a *Allocator) Free(mID int) int {
    count := 0
    for i := 0; i < len(a.memory); i++ {
        if a.memory[i] == mID {
            a.memory[i] = 0
            count++
        }
    }
    return count
}

func (a *Allocator) isMemoryFree(startIndex, size int) bool {
    for i := startIndex; i < startIndex+size; i++ {
        if a.memory[i] != 0 {
            return false
        }
    }
    return true
}

func (a *Allocator) assignMemory(startIndex, size, mID int) {
    for i := startIndex; i < startIndex+size; i++ {
        a.memory[i] = mID
    }
}

File: data/EXP01/python_2/2502/2502.output_2.py
class Allocator:

    def __init__(self, n: int):
        self.memory = [0] * n
        self.block_dict = {}
    
    def allocate(self, size: int, mID: int) -> int:
        for i in range(len(self.memory)):
            if self.memory[i] == 0:
                available = True
                for j in range(size):
                    if i+j >= len(self.memory) or self.memory[i+j] != 0:
                        available = False
                        break
                if available:
                    for j in range(size):
                        self.memory[i+j] = mID
                    if mID in self.block_dict:
                        self.block_dict[mID].append(i)
                    else:
                        self.block_dict[mID] = [i]
                    return i
        return -1
    
    def free(self, mID: int) -> int:
        if mID in self.block_dict:
            count = 0
            for block in self.block_dict[mID]:
                for i in range(block, len(self.memory)):
                    if self.memory[i] == mID:
                        self.memory[i] = 0
                        count += 1
                    else:
                        break
            del self.block_dict[mID]
            return count
        return 0
